In [2]:
import pandas as pd
import numpy as np 
from scipy.stats import ttest_ind
from statsmodels.stats.weightstats import ztest


In [95]:
df = pd.read_csv('D:\Learning\Machine-Learning-\data\dataset_1.csv')
df.head(10)

Rank            City Population Area KM2 Area   M2 Density KM2 Density  M2  \
0   1.0          Manila  1,846,513       43        17      43,062     111,532   
1   2.0     Mandaluyong    425,758       11         4      38,495      99,703   
2   3.0         pateros     65,227        2         1      37,061      95,988   
3   4.0        Caloocan  1,661,584       53        21      31,233      80,893   
4   5.0        ben fren    212,646        7         3      30,001      77,702   
5   6.0        Katmandu  1,442,000       49        19      29,161      75,526   
6   7.0           Dakka  8,906,136      306       118      29,069      75,289   
7   8.0          Makati    629,616       22         8      28,975      75,044   
8   9.0       Manhattan  1,628,701       59        23      27,544      71,340   
9  10.0  Port-au-Prince    987,310       36        14      27,395      70,953   

                        Country                Year  
0                  Filipinler *  2020  [1] 2018 [2]  
1                  Filipinler *       2020  [1] [3]  
2                  Filipinler *  2020  [1] 2013 [3]  
3                  Filipinler *       2020  [1] [3]  
4                      İsrail *        2022 [4] [5]  
5                       Nepal *            2020 [6]  
6                   Bangladeş *                   -  
7                  Filipinler *       2020  [1] [3]  
8   Amerika Birleşik Devletleri                2020  
9                       Hatay *            2015 [7]

In [36]:
#Task 1: Rearrange and print out top 10 cities with most populations and top 10 cities with smallest population
top_10_city_most_populations = df.sort_values(by='Population',ascending=True).head(10)
print(top_10_city_most_populations[['City','Population']])

                     City  Population
31                  naber   1,072,161
55  Ciudad Nezahualcóyotl   1,110,565
5                Katmandu   1,442,000
8               Manhattan   1,628,701
68              Barselona   1,636,732
3                Caloocan   1,661,584
23                    Şam   1,711,000
0                  Manila   1,846,513
69                Cakarta  10,560,000
28              Kallithea     100,050


In [37]:
top_10_city_smallest_population = df.sort_values(by='Population',ascending=False).head(10)
print(top_10_city_smallest_population[['City','Population']])

                   City Population
9        Port-au-Prince    987,310
60  Sint-Jans-Molenbeek     97,462
25               Asmara    963,000
61                 Seul  9,962,393
71               Kahire  9,539,673
54   Asnières-sur-Seine     85,191
41           Courbevoie     81,719
13                pasig    803,159
58        Modi'in Illit     80,894
6                 Dakka  8,906,136


In [38]:
#Task 2: Print out name of the country with at least 3 city
country_city_count = df['Country'].value_counts()
countries_with_at_least_3_cities = country_city_count[country_city_count >= 3]
print(countries_with_at_least_3_cities)


Country
 Filipinler *                     18
 Fransa *                         17
 Belçika *                         5
 İspanya *                         4
 Amerika Birleşik Devletleri *     3
 Endonezya *                       3
 İsrail *                          3
 Yunanistan *                      3
 Hindistan *                       3
Name: count, dtype: int64


In [39]:
#Task 3: Print out top 5 country with the most cities appear
top_5_country_with_most_cities_appear = countries_with_at_least_3_cities.sort_values(ascending=False).head(5)
print(top_5_country_with_most_cities_appear)

Country
 Filipinler *                     18
 Fransa *                         17
 Belçika *                         5
 İspanya *                         4
 Amerika Birleşik Devletleri *     3
Name: count, dtype: int64


In [104]:
df['Population'] = pd.to_numeric(df['Population'], errors='coerce')
df['Area KM2'] = pd.to_numeric(df['Area KM2'], errors='coerce')

In [105]:
#Task 4: Print out city with population and area in top 20 
top_20_population = df['Population'].nlargest(20).min()
top_20_area = df['Area KM2'].nlargest(20).min()
top_20_population_area = df[(df['Population'] >= top_20_population) & (df['Area KM2'] >= top_20_area)]
print(top_20_population_area[['City', 'Population', 'Area KM2']])

Empty DataFrame
Columns: [City, Population, Area KM2]
Index: []


In [ ]:
# Task 5: Thống kê mật độ dân số theo quốc gia

nan

In [ ]:
# Task 6: Thông kê các thành phố có dân số lớn nhất của từng quốc gia (chỉ tính những quốc gia có 2 thành phố xuất hiện trở lên trong bảng)

In [4]:
df2 = pd.read_csv('D:\Learning\Machine-Learning-\data\dataset_2.csv')
df2.head(5)

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [ ]:
# Task 1 : In ra index của 10 học sinh có math score và reading score đều nằm trong top 100 
top_10_mathscore_readingscore = df2[(df2['rank_math_score'] <= 100) & (df2['rank_reading_score'] <= 100)] # Combine two condition together 
print(top_10_mathscore_readingscore.index[:10]) # print out the index 

Index([7, 17, 18, 33, 55, 59, 61, 66, 75, 76], dtype='int64')


In [ ]:
df2['rank_reading_score']= df2['reading score'].rank(method='max') 
df2['rank_math_score'] = df2['math score'].rank(method='max')

In [91]:
# Task 2: In ra index của 20 học sinh có điểm writing score cao nhất của mỗi group(race/ethnicity)
top_20_per_group = df2.groupby('race/ethnicity')['writing score'].nlargest(20)
# Extract the indices
top_20_indices = top_20_per_group.index
# Print the indices
print("Indices of the top 20 students with highest writing scores for each group:")
print(top_20_indices.tolist())

Indices of the top 20 students with highest writing scores for each group:
[('group A', 546), ('group A', 614), ('group A', 571), ('group A', 820), ('group A', 983), ('group A', 702), ('group A', 539), ('group A', 623), ('group A', 468), ('group A', 489), ('group A', 778), ('group A', 911), ('group A', 77), ('group A', 300), ('group A', 423), ('group A', 837), ('group A', 378), ('group A', 490), ('group A', 651), ('group A', 170), ('group B', 855), ('group B', 514), ('group B', 2), ('group B', 6), ('group B', 94), ('group B', 121), ('group B', 501), ('group B', 474), ('group B', 919), ('group B', 577), ('group B', 833), ('group B', 551), ('group B', 722), ('group B', 158), ('group B', 667), ('group B', 715), ('group B', 803), ('group B', 815), ('group B', 125), ('group B', 749), ('group C', 165), ('group C', 594), ('group C', 717), ('group C', 334), ('group C', 347), ('group C', 381), ('group C', 397), ('group C', 229), ('group C', 325), ('group C', 732), ('group C', 979), ('group C', 

In [34]:
# Task 3: Thống kê số lượng học sinh của group A theo parental level of education
groupA_parental_level_of_education_students = df2.groupby(df2['race/ethnicity'] == 'group A').value_counts()
print(groupA_parental_level_of_education_students)

race/ethnicity  gender  race/ethnicity  parental level of education  lunch         test preparation course  math score  reading score  writing score  avg_score
False           female  group B         associate's degree           free/reduced  completed                68          77             80             75.000000    1
                male    group C         some college                 standard      completed                93          84             90             89.000000    1
                                        master's degree              standard      none                     71          67             67             68.333333    1
                                                                                                            79          72             69             73.333333    1
                                                                                                                        78             77             78.000000    1
               

Base on the answer we can conclude that the relation between education levels with average scores of children is yes. The higher educational levels of parents the higher average score of children

In [43]:
# Task 4: Có phải xu hướng chung, cha mẹ có học vấn càng cao thì điểm số trung bình 3 môn của con cái cũng càng cao không? 
df2['avg_score'] = (df2['math score'] + df2['reading score'] + df2['writing score']) / 3
groups_data = df2.groupby(['parental level of education'])['avg_score'].mean()
ans = groups_data.sort_values(ascending=False)
print(ans)

parental level of education
master's degree       73.598870
bachelor's degree     71.923729
associate's degree    69.569069
some college          68.476401
some high school      65.108007
high school           63.096939
Name: avg_score, dtype: float64


If we use normal statistics we can not conclude the answer for the question whether the lunch affected to average score of students. We should use Hypothesis Testing right here, with critical values is 0.05, H0 is lunch not affected to avg score of students

In [45]:
# Task 5: Chất lượng bữa ăn có tương phản với điểm số trung bình 3 môn của hs không? 
groups_lunch_data = df2.groupby(['lunch'])['avg_score'].mean()
ans_task_5 = groups_lunch_data.sort_values(ascending=False)
print(ans_task_5)

lunch
standard        70.837209
free/reduced    62.199061
Name: avg_score, dtype: float64


In [52]:
# Scores for each lunch type
standard_scores = df2[df2['lunch'] == 'standard']['avg_score']
reduced_scores = df2[df2['lunch'] == 'free/reduced']['avg_score']
# Perform a z-test
z_stat, p_value = ztest(standard_scores, reduced_scores)
print(f"Z-statistic: {z_stat}, P-value: {p_value}")
if p_value < 0.05:
    print("Lunch is not affected to average score of students")
else:
    print("Lunch is affected to average score of students")

Z-statistic: 9.575113051511476, P-value: 1.0174593231172428e-21
Lunch is not affected to average score of students


In [54]:
# Task 6: Tìm ra top 10 học sinh có điểm toán cao nhất của mỗi group 
group_students = df2.groupby(['race/ethnicity'])['math score'].nlargest(10)
print(group_students)

race/ethnicity     
group A         623    100
                539     97
                546     92
                571     91
                702     87
                820     85
                614     82
                300     81
                77      80
                444     79
group B         855     97
                501     94
                815     94
                121     91
                919     91
                2       90
                333     90
                474     90
                6       88
                146     88
group C         104     98
                934     98
                864     97
                165     96
                562     96
                717     96
                612     94
                710     93
                594     92
                736     92
group D         625    100
                712     98
                179     97
                618     95
                903     93
                957     92
        

The answer for task 7 is yes, the avg score of students who completed test preparation course is higher than students not completed it, with the same group,level of education of parent,lunch. This can explained by the effort of the students for training.

In [56]:
# Task 7: Với những học sinh ở cùng 1 group, cùng parental level of education, cùng lunch, việc tham gia test preapration course có giúp học sinh đó có điểm trung bình 3 môn cao hơn những học sinh khác không? 
group_data_task7 = df2.groupby(['race/ethnicity','parental level of education','lunch','test preparation course'])['avg_score'].mean()
print(group_data_task7)

race/ethnicity  parental level of education  lunch         test preparation course
group A         associate's degree           free/reduced  completed                  65.166667
                                                           none                       56.133333
                                             standard      completed                  74.000000
                                                           none                       65.333333
                bachelor's degree            free/reduced  completed                  55.666667
                                                                                        ...    
group E         some college                 standard      none                       71.283333
                some high school             free/reduced  completed                  73.333333
                                                           none                       57.000000
                                             standard

In [2]:
df3 = pd.read_csv('D:\Learning\Machine-Learning-\data\dataset_3.csv')
df3.head()

title  \
0  2020 Summer Vintage Flamingo Print  Pajamas Se...   
1  SSHOUSE Summer Casual Sleeveless Soirée Party ...   
2  2020 Nouvelle Arrivée Femmes Printemps et Été ...   
3  Hot Summer Cool T-shirt pour les femmes Mode T...   
4  Femmes Shorts d'été à lacets taille élastique ...   

                                          title_orig  price  retail_price  \
0  2020 Summer Vintage Flamingo Print  Pajamas Se...  16.00            14   
1   Women's Casual Summer Sleeveless Sexy Mini Dress   8.00            22   
2  2020 New Arrival Women Spring and Summer Beach...   8.00            43   
3  Hot Summer Cool T Shirt for Women Fashion Tops...   8.00             8   
4  Women Summer Shorts Lace Up Elastic Waistband ...   2.72             3   

  currency_buyer  units_sold  uses_ad_boosts  rating  rating_count  \
0            EUR         100               0    3.76            54   
1            EUR       20000               1    3.45          6135   
2            EUR         100               0    3.57            14   
3            EUR        5000               1    4.03           579   
4            EUR         100               1    3.10            20   

   rating_five_count  ...  merchant_rating_count  merchant_rating  \
0               26.0  ...                    568         4.128521   
1             2269.0  ...                  17752         3.899673   
2                5.0  ...                    295         3.989831   
3              295.0  ...                  23832         4.020435   
4                6.0  ...                  14482         4.001588   

                merchant_id  merchant_has_profile_picture  \
0  595097d6a26f6e070cb878d1                             0   
1  56458aa03a698c35c9050988                             0   
2  5d464a1ffdf7bc44ee933c65                             0   
3  58cfdefdacb37b556efdff7c                             0   
4  5ab3b592c3911a095ad5dadb                             0   

   merchant_profile_picture                                      product_url  \
0                       NaN  https://www.wish.com/c/5e9ae51d43d6a96e303acdb0   
1                       NaN  https://www.wish.com/c/58940d436a0d3d5da4e95a38   
2                       NaN  https://www.wish.com/c/5ea10e2c617580260d55310a   
3                       NaN  https://www.wish.com/c/5cedf17ad1d44c52c59e4aca   
4                       NaN  https://www.wish.com/c/5ebf5819ebac372b070b0e70   

                                     product_picture  \
0  https://contestimg.wish.com/api/webimage/5e9ae...   
1  https://contestimg.wish.com/api/webimage/58940...   
2  https://contestimg.wish.com/api/webimage/5ea10...   
3  https://contestimg.wish.com/api/webimage/5cedf...   
4  https://contestimg.wish.com/api/webimage/5ebf5...   

                 product_id   theme crawl_month  
0  5e9ae51d43d6a96e303acdb0  summer     2020-08  
1  58940d436a0d3d5da4e95a38  summer     2020-08  
2  5ea10e2c617580260d55310a  summer     2020-08  
3  5cedf17ad1d44c52c59e4aca  summer     2020-08  
4  5ebf5819ebac372b070b0e70  summer     2020-08  

[5 rows x 43 columns]

In [35]:
# Task 1: Liệt ke các sản phẩm có doanh thu lớn hơn 100.000 Euro
df3['Sales_total_price'] = df3['retail_price'] * df3['units_sold']
filtered = df3[df3['Sales_total_price'] > 100000]
ans_task_1 = filtered[['title','Sales_total_price']]
print(ans_task_1)

                                                  title  Sales_total_price
1     SSHOUSE Summer Casual Sleeveless Soirée Party ...             440000
6     Women Fashion Loose Lace Blouse Blouse V Neck ...             300000
9     Femmes d'été, plus la taille décontractée lâch...             110000
17    2018 New Fashion Women's Tops Sexy Strappy Sle...            2500000
18    Débardeur d'été 5 Couleurs Grandes Tailles été...             180000
...                                                 ...                ...
1548  Robe d'été surdimensionnée pour dames Robe lon...             870000
1557  fr.delaval.nl : Nouvelle mode d'été pour femme...            1520000
1558  Mode d'été pour dames Jumpsuit sans manches im...             110000
1560  Chemises d'été pour femmes Plus col en V sans ...             125000
1567  Combinaison sans manches pour femmes couleur u...             140000

[256 rows x 2 columns]


In [65]:
# Task 2: Liệt kê các sản phẩm có units_sold nằm trong top 100, đồng thời nằm trong nhóm 10% sản phẩm có rating cao nhất:
rank_units_sold = df3.nlargest(100,'units_sold')
rating_count = df3['rating'].quantile(0.9)
ans_task_2 = df3[df3['rating'] > rating_count]
print(ans_task_2[['title','rating']])

                                                  title  rating
5     Plus la taille d'été femmes décontracté sans m...    5.00
31    Sweet Beading Cami Débardeurs pour femmes Spag...    4.37
61    Pantalon de couleur unie à la mode pour femmes...    5.00
74    Fashion Women Sleeveless Lace Vest Loose V Nec...    5.00
75    Nouvelle tendance col rond en forme de coeur t...    4.50
...                                                 ...     ...
1488  Nouvelles femmes d'été mode couleur unie short...    5.00
1497  Mode Ocean Wave Bague Argent et Or Rose Bague ...    4.48
1507  50 pcs mignon été étanche autocollants jaune b...    4.64
1528  Femmes Mode Lâche Casual Sans Manches Imprimé ...    4.67
1532  6 Color Plus Size Women's Sexy Lace Patchwork ...    5.00

[153 rows x 2 columns]


Correlation between rating and units_sold is approximate 0.0395 so we cannot conclude that high rating affected to units sold

In [ ]:
# Task 3: Có phải sản phẩm càng được rating nhiều thì càng bán được nhiều sản phẩm không
corr = np.corrcoef(df3['rating'],df3['units_sold'])
print(corr)

[[1.         0.03947825]
 [0.03947825 1.        ]]


In [90]:
# Task 4: Thống kê top 50 sản phẩm nằm trong rating từ 3.0 -> 4.0 bán chạy nhất
rating_from_3_to_4 = df3[(df3['rating'] >= 3) & (df3['rating'] <= 4)]
sort_unit_rating = rating_from_3_to_4.nlargest(50,'units_sold')
sort_unit_rating[['title','rating','units_sold']]

title  rating  units_sold
17    2018 New Fashion Women's Tops Sexy Strappy Sle...    3.83      100000
214   Nouveau Aeeival Femmes Vêtements À Manches Lon...    3.76      100000
251   T-shirt à manches courtes en mousseline de soi...    3.98      100000
1018  Femmes dentelle manches courtes hauts hauts ch...    3.82      100000
1168  Nouvelle arrivée femmes été sexy robe de soiré...    3.53      100000
6     Women Fashion Loose Lace Blouse Blouse V Neck ...    3.84       50000
333   Été Femme Chiffon Chiffon Couleur unie Col en ...    3.93       50000
381   Sexy Thong Cheeky Bottom Femmes Beachwear Brés...    3.45       50000
401   Femmes dentelle plage boho maxi robe d'été san...    3.94       50000
498   Fashion Women Back Deep V V Sexy Gilet sans do...    3.35       50000
510   Women Fashion Sexy Sleeveless Crop Cops en tri...    3.81       50000
790   Women's Fashion - Chemise sans manches à encol...    4.00       50000
893   Femmes sexy Boho à capuche moulante sans manch...    3.69       50000
952   Hot Dernières Sexy Bikini Sexy Bikini Femmes M...    3.83       50000
1028  Maillots de bain à rayures pour dames avec dou...    3.91       50000
1466  Summer Women Sexy White Broderie Sexy White De...    3.60       50000
1     SSHOUSE Summer Casual Sleeveless Soirée Party ...    3.45       20000
14    2019 Summer Women's Fashion Nouvelle robe sans...    3.65       20000
18    Débardeur d'été 5 Couleurs Grandes Tailles été...    3.82       20000
21    Gilet d'été pour femmes Gilet de mode couleur ...    3.60       20000
87    Femmes 2019 Mode Bandage de mode 2019 Maillot ...    4.00       20000
93    COMENII Femmes Tuniques d'été Tuniques Robes T...    3.70       20000
127   NOUVEAU Femmes Casual Débardeurs Blouses O-cou...    3.99       20000
144   Nouvelle mode femmes sans manches à carreaux D...    3.89       20000
149   Maillots de bain Femmes Triangle Bikini Set Ba...    3.96       20000
190   Robes de bohème de mode pour femmes Décoration...    3.61       20000
250   5 Couleurs Femmes Mode Shorts De Sport Couleur...    3.85       20000
256   Débardeur " Je ne suis pas parfait mais je sui...    3.96       20000
277   2019 nouvelle mode femmes mince plus la taille...    3.69       20000
307   Women's Sleeveless Sleeveless Deep V-neckage P...    3.94       20000
402   Women's Fashion Lace-up Solid Summer Shorts Da...    3.59       20000
431   2018 Plus Size Summer Women Fashion Fashion Se...    3.93       20000
446   2019 Summer Women Spaghetti Strap Mini Robe Fe...    3.33       20000
455   Mayzero New Summer Femmes Robe Long Beach Boho...    3.92       20000
472   2019 nouvelles femmes mode pantalon lâche tail...    3.64       20000
473   Robe sans manches à imprimé floral Élégante et...    3.71       20000
478   2017 Nouvelles Femmes Dames D'été Soild Couleu...    3.66       20000
483   XS-8XL Robes Plus La Taille Vêtements De Mode ...    3.62       20000
500   Sexy Women Jupe robe en dentelle Bodycon Jupe ...    3.39       20000
516   Femmes Couleur unie Taille élastiquée Taille d...    3.95       20000
533   Femmes mode cocktail soirée maxi robe en mouss...    3.37       20000
547   Femmes deux pièces ensemble imprimé vague rayé...    3.41       20000
562   Robe décontractée d'été pour femme Mini-robe c...    3.88       20000
563   Robe d'été haute qualité plus taille 6XL 2016 ...    3.73       20000
564   Robe d'été longue Robe d'été longue Robe de pl...    3.53       20000
608   Femmes robe d'été sexy Boho Maxi longue robe d...    3.42       20000
630   Maillot de bain Bikini Set Rembourré Bandage B...    3.74       20000
651   Nouvelles femmes mode couleur unie Shorts de s...    3.60       20000
654   Pantalon de mode d'été Femmes Leggings Pantalo...    3.76       20000
665   NOUVELLES femmes lâche couleur unie robe à man...    3.63       20000

There are 224 items with retail price greater than 50

In [ ]:
# Task 5: Có bao nhiêu sản phẩm có giá bán lớn hơn 50
count_item_price = df3.value_counts([df3['retail_price'] > 50])
print(count_item_price)

retail_price
False           1349
True             224
Name: count, dtype: int64


In [3]:
df4 = pd.read_csv('D:\Learning\Machine-Learning-\data\ds_salaries.csv')
df4.head(10)

Unnamed: 0  work_year experience_level employment_type  \
0           0       2020               MI              FT   
1           1       2020               SE              FT   
2           2       2020               SE              FT   
3           3       2020               MI              FT   
4           4       2020               SE              FT   
5           5       2020               EN              FT   
6           6       2020               SE              FT   
7           7       2020               MI              FT   
8           8       2020               MI              FT   
9           9       2020               SE              FT   

                    job_title    salary salary_currency  salary_in_usd  \
0              Data Scientist     70000             EUR          79833   
1  Machine Learning Scientist    260000             USD         260000   
2           Big Data Engineer     85000             GBP         109024   
3        Product Data Analyst     20000             USD          20000   
4   Machine Learning Engineer    150000             USD         150000   
5                Data Analyst     72000             USD          72000   
6         Lead Data Scientist    190000             USD         190000   
7              Data Scientist  11000000             HUF          35735   
8       Business Data Analyst    135000             USD         135000   
9          Lead Data Engineer    125000             USD         125000   

  employee_residence  remote_ratio company_location company_size  
0                 DE             0               DE            L  
1                 JP             0               JP            S  
2                 GB            50               GB            M  
3                 HN             0               HN            S  
4                 US            50               US            L  
5                 US           100               US            L  
6                 US           100               US            S  
7                 HU            50               HU            L  
8                 US           100               US            L  
9                 NZ            50               NZ            S

In [30]:
# Task 1: Job title nao` co muc luong theo USD cao nhat? 
avg_salary_in_USD = df4.groupby(by='job_title')['salary_in_usd'].mean()
job_title_highest_avg_salary_in_USD = avg_salary_in_USD.idxmax()
highest_salary = avg_salary_in_USD.max()
print(print(f"Job title with the highest average salary: {job_title_highest_avg_salary_in_USD} {highest_salary:.1f}"))

Job title with the highest average salary: Data Analytics Lead 405000.0
None


In [60]:
# Task 2: DS mang quoc tich nao co muc luong trung binh cao nhat?
DS = df4[df4['job_title'] == "Data Scientist"]
DS_residence = DS.groupby(by='employee_residence')['salary_in_usd'].mean()
highest_DS_residence = DS_residence.idxmax()
highest_DS_residence_salary = DS_residence.max()
print(f"Data Scientist from: {highest_DS_residence} has the salary ${highest_DS_residence_salary:.2f} is highest")

Data Scientist from: US has the salary $149408.33 is highest


The answer for question 3 is foreign employees is not earns higher than domestic employee. The average salary of US employees ($149194.1) is much higher than foreign employees ($67754.0)

In [27]:
# Task 3: Co phai o USA, nhan su ngoai quoc nhan dc muc luong trung binh cao hon nhan su trong nuoc khong?
# Filter USA employees
employee_USA = df4[df4['employee_residence'] == "US"]
avg_salary_employee_USA = employee_USA['salary_in_usd'].mean()
# Filter foreign employees
employee_not_USA = df4[df4['employee_residence'] != "US"]
avg_salary_employee_not_USA = employee_not_USA['salary_in_usd'].mean()
print(f"Average salary for domestic employees: {avg_salary_employee_USA:.1f}")
print(f"Average salary for foreign employees: {avg_salary_employee_not_USA:.1f}")

Average salary for domestic employees: 149194.1
Average salary for foreign employees: 67754.0


In [53]:
print(DE[DE['company_size'] == 'S']['salary_in_usd'].describe())
print(DE[DE['company_size'] == 'M']['salary_in_usd'].describe())

count         3.000000
mean     145000.000000
std       25980.762114
min      115000.000000
25%      137500.000000
50%      160000.000000
75%      160000.000000
max      160000.000000
Name: salary_in_usd, dtype: float64
count        47.000000
mean     142488.702128
std       43432.335145
min       25000.000000
25%      115000.000000
50%      140250.000000
75%      165200.000000
max      243900.000000
Name: salary_in_usd, dtype: float64


Average Salary of job title contains Data Engineer in USA with company size is small and medium are: 
- Small Size: 145000
- Medium Size: 142488.7

In [ ]:
# Task 4: Muc luong trung binh cua DE (SE level) o cac cong ty nho va vua o USA
DE = df4[(df4['job_title'].str.contains("Data Engineer") & (df4['company_location'] == "US") & (df4['experience_level'] == "SE"))]
avg_DE_SE_salary = DE.groupby(by='company_size')['salary_in_usd'].mean()
avg_DE_SE_salary

company_size
L    172550.000000
M    142488.702128
S    145000.000000
Name: salary_in_usd, dtype: float64

In [4]:
# Task 5: So sanh muc luong cua DS o USA o cac muc trinh do (experience_level) khac nhau
DS_in_USA = df4[df4['job_title'].str.contains("Data Scientist") & (df4['company_location'] == "US")]
avg_salary_DS_in_USA = DS_in_USA.groupby(by='experience_level')['salary_in_usd'].mean()
avg_salary_DS_in_USA

experience_level
EN     88833.333333
EX    416000.000000
MI    115618.111111
SE    169186.915254
Name: salary_in_usd, dtype: float64

In [ ]:
# Task 6: Nhan su la nguoi nuoc ngoai chiem ti le bao nhieu phan tram o cac cong ty lon o USA


In [ ]:
# Task 7: Tim ra 3 job title co muc luong trung binh cao nhat o level MI danh cho nhan su trong nuoc

In [ ]:
# Task 8: Job nao o Germany (DE) co muc luong trung binh thap nhat?

In [ ]:
# Task 9: Job nao co khoang cach thu nhap giua SE va MI it nhat?  